### PR TITLE
feat: Use `satisfies` for feature names list

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves [] 

### Description
This PR adds a type check for feature names thanks to typescript's `satisfies`
We also needed to bump the `metro-react-native-babel-preset` to support `satisfies` in the metro bundler

### Screenshots
**Before**
Screenshot 2024-01-24 at 14 09 26

**After**
Screenshot 2024-01-24 at 14 09 07

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Update features type - mounri+sultan

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Update features type - mounri+sultan)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 33bbf3f381f28da12d671a28ed5fe4c48441c9a5
</p>

**Review by MounirDhahri — APPROVED** [review]
🍿🍿🍿

File: ios/Artsy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4534,11 +4534,7 @@
 					"-Werror=return-type",
 				);
 				OTHER_CPLUSPLUSFLAGS = "$(inherited)";
-				OTHER_LDFLAGS = (
-					"$(inherited)",
-					"-Wl",
-					"-ld_classic",
-				);
```
Comment (gkartalis) [inline ios/Artsy.xcodeproj/project.pbxproj:4541]: when pulling and installing everything (yarn install:all) this diff (whole file) comes back, I would suggest to remove it 🤔 not sure how it ended up here

**Review by gkartalis — APPROVED** [review]
one comment about removing the project.pbxproj diff but other than that looks awesome!

**Review by gkartalis — APPROVED** [review]
🛳️

File: ios/Artsy.xcodeproj/project.pbxproj (hunk 1 above)
Comment (MounirDhahri) [reply ios/Artsy.xcodeproj/project.pbxproj]: I made a PR for these changes here https://github.com/artsy/eigen/pull/9783

**Pull request merged**